docs/general/run-a-node/prerequisites/system-configuration.md

```diff
@@ -57,3 +60,132 @@ If you are running Oasis Node via [Docker](https://www.docker.com/) you can pass
 --ulimit nofile=102400:1048576
 ```
 
+## Running Oasis Services with Non-root System User {#non-root}
+
+:::caution
+
+Beginning with **Oasis Core 22.1.x release series** it is
+**no longer allowed to run Oasis Node** (i.e. the `oasis-node` binary)
+**as root** (effective user ID of 0).
+
+:::
+
+Running network accessible services as the root user is extremely bad for
+system security as a general rule. While it would be "ok" if we could drop
+privileges, `syscall.AllThreadsSyscall` does not work if the binary uses `cgo`
+at all.
+
+Nothing in Oasis Node will ever require elevated privileges.
+Attempting to run the `oasis-node` process as the root user will now terminate
+immediately on startup.
+
+While there may be specific circumstances where it is safe to run network
+services with the effective user ID set to 0, the overwhelming majority of cases
+where this is done is a misconfiguration.
+
+### Changing Your Setup to Run Oasis Services with Non-root System User {#change-to-non-root}
+
+:::info
+In these examples, we change the setup to run Oasis Services (e.g. Oasis Node)
+with a non-root system user named `oasis`.
+These instructions also assume that the node's datadir is `/serverdir/node`.
+
+Adjust these as appropriate to your setup.
+:::
+
+1. Create the `oasis` system user:
+
+<Tabs>
+<TabItem value="Ubuntu">
+
+As root, run:
+
+```bash
+adduser --system oasis --shell /usr/sbin/nologin
+```
+
+</TabItem>
+<TabItem value="Fedora">
+
+As root, run:
+```bash
+useradd -r -s /usr/sbin/nologin
+```
+
+</TabItem>
+
+<TabItem value="Ansible">
+
+Add the following task to your playbook:
+
+```yml
+- name: Create oasis user
+  user:
+    name: oasis
+    comment: Oasis Services user
+    system: yes
+    shell: /usr/sbin/nologin
+```
+
+</TabItem>
+</Tabs>
+
+:::tip
+
+Setting `oasis` user's Shell to `/usr/sbin/nologin` prevents (accidentally)
+logging-in as this user.
+
+:::
+
+2. Stop your Oasis Node.
+
+3. Transfer ownership of the datadir to the `oasis` user:
+
+```sh
+chown -R oasis /serverdir/node
+```
+
+See https://github.com/oasisprotocol/docs/blob/main/docs/general/run-a-node/troubleshooting.md#invalid-permissions for more information.
```

Comment:
https://docs.oasis.dev/general/run-a-node/troubleshooting#invalid-permissions

Reply:
Yeah, one just links to the docs via internal path.